examples/dao/assets/dao-fund-lsig.py

```diff
@@ -0,0 +1,42 @@
+from pyteal import *
+
+def proposal_lsig():
+    """
+    Represents DAO treasury (ALGO/ASA)
+    """
+
+    # check no rekeying, close remainder to, asset close to for a txn
+    def basic_checks(txn: Txn): return And(
+        txn.rekey_to() == Global.zero_address(),
+        txn.close_remainder_to() == Global.zero_address(),
+        txn.asset_close_to() == Global.zero_address()
+    )
+
+    # ideally we should retrict the sender here(so that anyone cannot withdraw funds)
+    receive_or_send = basic_checks(Txn)
+
+    # verify funds are transfered only when paired with DAO app (during execute call)
+    payment = And(
+        # verify first transaction
+        basic_checks(Gtxn[0]),
+        Gtxn[0].type_enum() == TxnType.ApplicationCall,
+        Gtxn[0].application_id() == Tmpl.Int("TMPL_DAO_APP_ID"),
+        Gtxn[0].application_args[0] == Bytes("execute"),
+
+        # verify second transaction (either payment in asa or ALGO)
+        basic_checks(Gtxn[1]),
+        Or(
+            Gtxn[1].type_enum() == TxnType.AssetTransfer,
+            Gtxn[1].type_enum() == TxnType.Payment,
+        )
+    )
+
+    program = program = Cond(
+        [Global.group_size() == Int(1), receive_or_send],
+        [Global.group_size() == Int(2), payment]
+    )
```

Comment:
we need to restrict the access to the account:
* anyone can send to the dao-fund
* anyone can opt-in 
* withdraw (payment) - as below
* other transactions must be forbidden

Reply:
We're following all these points (except that anyone can withdraw funds). We can add a hardcoded address for withdrawal.

Reply:
For withdraw:
* we don't restrict recipient (transfer can go anywhere)
* but the transaction must be in a group transaction with the `dao_app execute` call.

Reply:
got it, so we're doing this as well (in the `payment` branch), where we ensure funds are only transferred when paired with the `execute` call.